inference-engine/tests/functional/plugin/cpu/single_layer_tests/roialign.cpp

```diff
@@ -3,108 +3,172 @@
         ROIAlignSpecificParams roiPar;
-        std::tie(roiPar, netPr, td) = basicParamsSet;
-        std::tie(pooledH, pooledW, spatialScale, samplingRatio,
-                 proposal, mode, inputShape) = roiPar;
+        std::tie(roiPar, netPrecision, td) = basicParamsSet;
+
+        int pooledH;
+        int pooledW;
+        float spatialScale;
+        int samplingRatio;
+        std::string mode;
+        ROIAlignShapes inputShapes;
+        std::tie(pooledH, pooledW, spatialScale, samplingRatio, mode, inputShapes) = roiPar;
         std::ostringstream result;
-        result << "ROIAlignTest_";
-        result << std::to_string(obj.index);
+
+        result << netPrecision << "_IS=";
+        for (const auto& shape : inputShapes) {
+            result << CommonTestUtils::partialShape2str({ shape.first }) << "_";
+        }
+        result << "TS=";
+        for (const auto& shape : inputShapes) {
+            result << "(";
+            if (!shape.second.empty()) {
+                auto itr = shape.second.begin();
+                do {
+                    result << CommonTestUtils::vec2str(*itr);
+                } while (++itr != shape.second.end() && result << "_");
+            }
+            result << ")_";
+        }
+
         result << "pooledH=" << pooledH << "_";
         result << "pooledW=" << pooledW << "_";
         result << "spatialScale=" << spatialScale << "_";
         result << "samplingRatio=" << samplingRatio << "_";
-        result << (netPr == Precision::FP32 ? "FP32" : "BF16") << "_";
         result << mode << "_";
         result << CPUTestsBase::getTestCaseName(cpuParams);
+
         return result.str();
     }
 protected:
+    void generate_inputs(const std::vector<ngraph::Shape>& targetInputStaticShapes) override {
+        inputs.clear();
+        const auto& funcInputs = function->inputs();
+
+        ov::runtime::Tensor data_tensor;
+        const auto& dataPrecision = funcInputs[0].get_element_type();
+        const auto& dataShape = targetInputStaticShapes.front();
+        data_tensor = ov::test::utils::create_and_fill_tensor(dataPrecision, dataShape, 10, 0, 1000);
+
+        const auto& coordsET = funcInputs[1].get_element_type();
+        auto coordsTensor = ov::runtime::Tensor{ coordsET, targetInputStaticShapes[1] };
+        if (coordsET == ElementType::f32) {
+            auto coordsTensorData = static_cast<float*>(coordsTensor.data());
+            for (size_t i = 0; i < coordsTensor.get_size(); i += 4) {
+                coordsTensorData[i] = 1.f;
+                coordsTensorData[i] = 1.f;
+                coordsTensorData[i] = 19.f;
+                coordsTensorData[i] = 19.f;
+            }
+        } else if (coordsET == ElementType::bf16) {
+            auto coordsTensorData = static_cast<std::int16_t*>(coordsTensor.data());
+            for (size_t i = 0; i < coordsTensor.get_size(); i += 4) {
+                coordsTensorData[i] = static_cast<std::int16_t>(ngraph::float16(1.f).to_bits());
```

Comment:
as I can see we used some different approach to generate `bf16` values.
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/ie_test_utils/common_test_utils/data_utils.hpp#L259

where did you take this approach?

Reply:
BF16 data generations was fixed, thank you!